ui/src/chat/ChatChannel.tsx

```diff
@@ -50,14 +51,26 @@ function ChatChannel({ title }: ViewProps) {
 
   const joinChannel = useCallback(async () => {
     setJoining(true);
-    await useChatState.getState().joinChat(chFlag);
+    if (joinRef) {
+      setTimeout(async () => {
+        await useChatState.getState().joinChat(chFlag);
+      }, 300);
```

Comment:
Is the 300 to allow for the modal to open? Will this be sufficient for lower end devices? 

Also, please extract to a constant value so we can easily tune it here and on line 68

Reply:
Actually we're waiting for the join to complete (on the new group/channel you're joining). For some reason we get the fact back that the group join completes (and the promise on the frontend returns true), but if we don't wait a bit before initializing the chat it'll fail. I had to do the same thing with notes. We need to figure out why that's happening on the backend, this is just a workaround for now.

Reply:
I thought we had figured this out, but having trouble remembering.

Reply:
That was a different instance, but I'm down to figure this out.